services/streammanager/kafka/kafkamanager_test.go

```diff
@@ -800,6 +838,17 @@ func TestSendMessage(t *testing.T) {
 	})
 }
 
+func TestPublish(t *testing.T) {
+	t.Run("no topic publish", func(t *testing.T) {
+		kafkaStats.publishTime = getMockedTimer(t, gomock.NewController(t))
+		p := &client.Producer{}
+		pm := &ProducerManager{p: p}
+		message := prepareMessage("", "userId", []byte(`{"test": "value"}`), time.Now())
+		err := publish(context.Background(), pm, message)
+		require.Equal(t, "no topic provided for message 0", err.Error())
```

Comment:
FYI, there is a `require.ErrorContains` helper too which takes an error instead of a string.

Reply:
This is addressed